documentation/developer-guide/modules/tooling-guide/pages/maven-plugin.adoc

```diff
@@ -0,0 +1,424 @@
+| `outputDirectory` | The path to the directory where the generated diagrams will be written to. | `String` | none | {ok}
+| `targetFormats` | A list formats in which the diagram(s) will be created. A diagram will be generated for each specified format. | `String` | none | {ok}
+|===
+
+=== Generating Sample JSON Payload
+
+The `generateJsonPayload` goal generates a valid sample JSON payload for a given Aspect Model as it could be returned by an Aspect that implements that Aspect Model. The default life cycle phase for the goal is `generate-resources`.
+
+Usage:
+
+[source,maven,subs=attributes+]
+----
+<build>
+  <plugins>
+     <plugin>
+        <artifactId>sds-aspect-model-maven-plugin</artifactId>
+        <executions>
+           <execution>
+              <id>generate-json-payload</id>
+              <goals>
+                 <goal>generateJsonPayload</goal>
+              </goals>
+           </execution>
+        </executions>
+        <configuration>
+           <aspectModelFilePath>{path-to-aspect-model/AspectModelFile.ttl}</aspectModelFilePath>
+           <outputDirectory>{directory-for-generated-source-files}</outputDirectory>
+        </configuration>
+     </plugin>
+  </plugins>
+</build>
+----
+
+Configuration Properties:
+
+[width="100%", options="header", cols="20,50,10,10,10"]
+|===
+| Property | Description | Type | Default Value | Required
+| `aspectModelFilePath` | The path to the Aspect Model file. | `String` | none | {nok}
+| `outputDirectory` | The path to the directory where the generated JSON payload will be written to. | `String` | none | {ok}
+|===
+
+== Aspect Model Migration
+
+The `migrate` goal migrates the given Aspect Model to the latest version of the meta model. The default life cycle phase for the goal is `initialize`.
+
+Usage:
+
+[source,maven,subs=attributes+]
+----
+<build>
+  <plugins>
+     <plugin>
+        <artifactId>sds-aspect-model-maven-plugin</artifactId>
+        <executions>
+           <execution>
+              <id>migrate-aspect-model</id>
+              <goals>
+                 <goal>migrate</goal>
+              </goals>
+           </execution>
+        </executions>
+        <configuration>
+           <aspectModelFilePath>{path-to-aspect-model/AspectModelFile.ttl}</aspectModelFilePath>
+           <outputDirectory>{directory-for-generated-source-files}</outputDirectory>
+        </configuration>
+     </plugin>
+  </plugins>
+</build>
+----
+
+Configuration Properties:
+
+[width="100%", options="header", cols="20,50,10,10,10"]
+|===
+| Property | Description | Type | Default Value | Required
+| `aspectModelFilePath` | The path to the Aspect Model file. | `String` | none | {nok}
+| `outputDirectory` | The path to the directory where the updated Aspect Model will be written to. | `String` | none | {ok}
+|===
+
+== Pretty Print
+
+The `prettyPrint` goal formats the given Aspect Model. The default life cycle phase for the goal is `generate-resources`.
```

Comment:
Does this overwrite the source file or does it write the formatted file to `target`? This should be explained. Should this be configurable? I could image use cases for both (although I personally would never want build tooling to overwrite source files).

Reply:
The formatted file is written to `target`.

I don't think it is necessary to explicitly have the option to override the source file. If a user really wants to do that, they could set the `target` to point to the location of the source file.